spring-boot-actuator/src/main/java/org/springframework/boot/actuate/info/InfoProvider.java

```diff
@@ -0,0 +1,17 @@
+package org.springframework.boot.actuate.info;
+
+import java.util.Map;
+
+/**
+ * information provider for the info endpoint
+ *
+ * @author Meang Akira Tanaka
+ */
+public interface InfoProvider {
+
+	/**
+	 * @return a collection of information
+	 */
+	Map<String, Object> provide();
```

Comment:
Break what interface? You just created it :)

If you return a strongly-typed object, it gives you the ability to evolve said object in the future without breaking anything. In general returning simple types like map from an interface is a bad idea. 

I am not too sure what to do at this point but it feels to me that the default implementation of `InfoProvider` should read the `info.*` keys from the context.